### ISSUE TITLE
[hardhat-ethers] Library linking hangs when a contract instance is used instead of an address

### ISSUE THREAD

**Issue by krasi-georgiev** [nomiclabs/hardhat#1037]
When running this test it just hangs when deploying the last contract

https://github.com/tellor-io/lens/blob/6b2e973e9adda48de80a7e7a6dee5401ff0be557/test/tests.js#L76

**Comment by scnale** [comment]
Oh, I'm not sure why it hangs but I'd suggest passing the address instead of the `ethers.Contract`.

```js
    fact = await ethers.getContractFactory(
      path.join("tellorcore", "contracts", "TellorGetters.sol:TellorGetters"),
      {
        libraries: {
          OldTellorTransfer: oldTransfer.address,
        }
      }
    );
```

**Comment by krasi-georgiev** [comment]
Shoot! yes that was it, apologies, user mistake thanks for the prompt reply

**Comment by fvictorio** [comment]
Re-opening this to see if we can throw a better error here.

**Comment by scnale** [comment]
As far as I can see, we were already throwing an error for non-address arguments that were passed to the linker. I added a test for this specific case in #1059 just to be sure in the future.

**Comment by fvictorio** [comment]
> As far as I can see, we were already throwing an error for non-address arguments that were passed to the linker.

I can reproduce this issue, so I don't think that's the case.

**Comment by fvictorio** [comment]
Oh, wait, it does show the correct message but it takes a looong time. I still think we should keep this open to improve that experience.

**Comment by fvictorio** [comment]
Fixed in `@nomiclabs/hardhat-ethers@2.0.2`.
